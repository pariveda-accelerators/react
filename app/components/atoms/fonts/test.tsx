import React from 'react';
import { render } from 'react-testing-library';
import * as Fonts from './';

describe('Fonts', () => {
  test('Renders Font Boldnesses', () => {
    const { container } = render(
      <>
        <Fonts.Body1 boldness={Fonts.BoldnessLevels.LIGHT}>
          Light Boldness
        </Fonts.Body1>
        <Fonts.Body1 boldness={Fonts.BoldnessLevels.MEDIUM}>
          Medium Boldness
        </Fonts.Body1>
        <Fonts.Body1 boldness={Fonts.BoldnessLevels.BOLD}>Bold Boldness</Fonts.Body1>
      </>,
    );
    expect(container).toMatchSnapshot();
  });
  test('Render Font Colors', () => {
    const { container } = render(
      <>
        <Fonts.Body1 color={Fonts.FontColor.WHITE}>White Font Color</Fonts.Body1>
        <Fonts.Body1 color={Fonts.FontColor.BLACK}>Black Font Color</Fonts.Body1>
      </>,
    );
    expect(container).toMatchSnapshot();
  });
  test('Renders Display Fonts', () => {
    const { container } = render(
      <>
        <Fonts.Display1>Display 1</Fonts.Display1>
        <Fonts.Display2>Display 2</Fonts.Display2>
        <Fonts.Display3>Display 3</Fonts.Display3>
      </>,
    );
    expect(container).toMatchSnapshot();
  });
  test('Renders Heading Fonts', () => {
    const { container } = render(
      <>
        <Fonts.H1>H 1</Fonts.H1>
        <Fonts.H2>H 2</Fonts.H2>
        <Fonts.H3>H 3</Fonts.H3>
      </>,
    );
    expect(container).toMatchSnapshot();
  });
  test('Renders Body Fonts', () => {
    const { container } = render(
      <>
        <Fonts.Body1>Body 1</Fonts.Body1>
        <Fonts.Body2>Body 2</Fonts.Body2>
        <Fonts.Body3>Body 3</Fonts.Body3>
      </>,
    );
    expect(container).toMatchSnapshot();
  });
  /** TODO - Include test once issue w/ theme colors is resolved */
  // test('Renders Link Fonts', () => {
  //   const { container } = render(
  //     <>
  //       <Fonts.Link1 linkUrl="">Link 1</Fonts.Link1>
  //       <Fonts.Link2 linkUrl="">Link 2</Fonts.Link2>
  //       <Fonts.Link3 linkUrl="">Link 3</Fonts.Link3>
  //     </>,
  //   );
  //   expect(container).toMatchSnapshot();
  // });
  test('Renders Label Fonts', () => {
    const { container } = render(
      <>
        <Fonts.Label>Label</Fonts.Label>
      </>,
    );
    expect(container).toMatchSnapshot();
  });
});
