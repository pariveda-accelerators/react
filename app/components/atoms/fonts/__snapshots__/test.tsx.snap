// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fonts Render Font Colors 1`] = `
.c1 {
  color: #3d4856;
  font-family: Courier;
  font-weight: 500;
  font-size: 16px;
  line-height: 20px;
}

.c0 {
  color: #f5f6fa;
  font-family: Courier;
  font-weight: 500;
  font-size: 16px;
  line-height: 20px;
}

<div>
  <div
    class="c0"
  >
    White Font Color
  </div>
  <div
    class="c1"
  >
    Black Font Color
  </div>
</div>
`;

exports[`Fonts Renders Body Fonts 1`] = `
.c0 {
  color: #3d4856;
  font-family: Courier;
  font-weight: 500;
  font-size: 16px;
  line-height: 20px;
}

.c1 {
  color: #3d4856;
  font-family: Courier;
  font-weight: 500;
  font-size: 14px;
  line-height: 18px;
}

.c2 {
  color: #3d4856;
  font-family: Courier;
  font-weight: 500;
  font-size: 12px;
  line-height: 16px;
}

<div>
  <div
    class="c0"
  >
    Body 1
  </div>
  <div
    class="c1"
  >
    Body 2
  </div>
  <div
    class="c2"
  >
    Body 3
  </div>
</div>
`;

exports[`Fonts Renders Display Fonts 1`] = `
.c0 {
  color: #3d4856;
  font-family: Courier;
  font-weight: 500;
  font-size: 72px;
  line-height: 80px;
}

.c1 {
  color: #3d4856;
  font-family: Courier;
  font-weight: 500;
  font-size: 48px;
  line-height: 54px;
}

.c2 {
  color: #3d4856;
  font-family: Courier;
  font-weight: 500;
  font-size: 40px;
  line-height: 46px;
}

<div>
  <div
    class="c0"
  >
    Display 1
  </div>
  <div
    class="c1"
  >
    Display 2
  </div>
  <div
    class="c2"
  >
    Display 3
  </div>
</div>
`;

exports[`Fonts Renders Font Boldnesses 1`] = `
.c0 {
  color: #3d4856;
  font-family: Courier;
  font-weight: 300;
  font-size: 16px;
  line-height: 20px;
}

.c1 {
  color: #3d4856;
  font-family: Courier;
  font-weight: 500;
  font-size: 16px;
  line-height: 20px;
}

.c2 {
  color: #3d4856;
  font-family: Courier;
  font-weight: 700;
  font-size: 16px;
  line-height: 20px;
}

<div>
  <div
    boldness="1"
    class="c0"
  >
    Light Boldness
  </div>
  <div
    boldness="2"
    class="c1"
  >
    Medium Boldness
  </div>
  <div
    boldness="3"
    class="c2"
  >
    Bold Boldness
  </div>
</div>
`;

exports[`Fonts Renders Heading Fonts 1`] = `
.c0 {
  color: #3d4856;
  font-family: Courier;
  font-weight: 500;
  font-size: 28px;
  line-height: 34px;
}

.c1 {
  color: #3d4856;
  font-family: Courier;
  font-weight: 500;
  font-size: 24px;
  line-height: 30px;
}

.c2 {
  color: #3d4856;
  font-family: Courier;
  font-weight: 500;
  font-size: 20px;
  line-height: 30px;
}

<div>
  <div
    class="c0"
  >
    H 1
  </div>
  <div
    class="c1"
  >
    H 2
  </div>
  <div
    class="c2"
  >
    H 3
  </div>
</div>
`;

exports[`Fonts Renders Label Fonts 1`] = `
.c0 {
  color: #3d4856;
  font-family: Courier;
  font-weight: 500;
  font-size: 11px;
  line-height: 14px;
}

<div>
  <div
    class="c0"
  >
    Label
  </div>
</div>
`;
